routes/shopItemById.js

```diff
@@ -15,20 +12,25 @@ shopItemById.post("/", (req, res, next) => {
 });
 
 shopItemById.delete("/", (req, res, next) => {
-    req.params.id = Number(req.params.id);
-    if (!isNaN(req.params.id)) {
-        const deleteIndex = req.database.shopItems.findIndex((item) => {
-            return item.id === req.params.id
-        })
-        if (deleteIndex !== -1) {
-            req.database.shopItems.splice(deleteIndex, 1)
-            utility.writeDatabase(req.database);
-            res.status(204).send();
+    try {
+        req.params.id = Number(req.params.id);
+        if (!isNaN(req.params.id)) {
+            db.run("DELETE FROM grocery_list WHERE id = $id", {
+                $id: req.params.id
+            }, function(err) {
+                if (err) {
+                    console.log(err);
+                    res.status(404).send();
```

Comment:
Deleting 0 items will not result in an error. err is not null only when there is a serious error (e.g. the disk is full, the database file got removed, etc). In this case code 500 should be returned.

You can use this.changes to see how many rows have been changed (i.e. deleted). If that is 0, than 404. Otherwise 204 should be returned.